contracts/lido/TokenRateNotifier.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-FileCopyrightText: 2024 Lido <info@lido.fi>
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity 0.8.10;
+
+import {IPostTokenRebaseReceiver} from "./interfaces/IPostTokenRebaseReceiver.sol";
+import {ITokenRateObserver} from "./interfaces/ITokenRateObserver.sol";
+import {ObserversArray} from "./ObserversArray.sol";
+
+/// @author kovalgek
+/// @notice Notifies all observers when rebase event occures.
+contract TokenRateNotifier is ObserversArray, IPostTokenRebaseReceiver {
```

Comment:
Why not include the factory for "general" token rate observer here?
I mean add method like:
```solidity
function addGeneralObserver(address lidoTokensBridge_, uint32 l2GasLimitForPushingTokenRate_) external {
        opStackTokenRateObserver = new OpStackTokenRateObserver(
            lidoTokensBridge_,
            l2GasLimitForPushingTokenRate_
        );

        emit OpStackTokenRateObserverDeployed(
            msg.sender,
            address(opStackTokenRateObserver),
            lidoTokensBridge_,
            l2GasLimitForPushingTokenRate_
        );
        
        addObserver(address(opStackTokenRateObserver));
}
```
And keep `addObserver` public to allow adding non-general observers.

This would simplify operations for adding a general token rate observer, although at the cost of deployment the contract upon Aragon Vote execution.

Reply:
As we agreed during the call, we can't create a general solution for the observer; thus, it isn't valid anymore.